code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -11,33 +11,40 @@ emp_act
 
 /mob/living/carbon/human/bullet_act(obj/item/projectile/P, def_zone)
 	if(!dna.species.bullet_act(P, src))
+		add_attack_logs(P.firer, src, "hit by [P.type] but got deflected by species '[dna.species]'")
 		return FALSE
 	if(P.is_reflectable)
 		var/can_reflect = check_reflect(def_zone)
-		if(can_reflect == 1) // proper reflection
-			visible_message("<span class='danger'>The [P.name] gets reflected by [src]!</span>", \
-						"<span class='userdanger'>The [P.name] gets reflected by [src]!</span>")
-			P.reflect_back(src)
-			return -1 // complete projectile permutation
+		var/reflected = FALSE
 
-		else if(can_reflect == 2) //If target is holding a toy sword
-			var/list/safe_list = list(/obj/item/projectile/beam/lasertag, /obj/item/projectile/beam/practice)
-			if(is_type_in_list(P, safe_list)) //And it's safe
-				visible_message("<span class='danger'>The [P.name] gets reflected by [src]!</span>", \
-				   "<span class='userdanger'>The [P.name] gets reflected by [src]!</span>")		
-				P.reflect_back(src)
-				return -1 // complete projectile permutation
+		switch(can_reflect)
+			if(1) // proper reflection
+				reflected = TRUE
+			if(2) //If target is holding a toy sword
+				var/static/list/safe_list = list(/obj/item/projectile/beam/lasertag, /obj/item/projectile/beam/practice)
+				reflected = is_type_in_list(P, safe_list) //And it's safe
 
+		if(reflected)
+			visible_message("<span class='danger'>The [P.name] gets reflected by [src]!</span>", \
+				   "<span class='userdanger'>The [P.name] gets reflected by [src]!</span>")
+			add_attack_logs(P.firer, src, "hit by [P.type] but got reflected")
+			P.reflect_back(src)
+			return -1
 
 	//Shields
 	if(check_shields(P, P.damage, "the [P.name]", PROJECTILE_ATTACK, P.armour_penetration))
 		P.on_hit(src, 100, def_zone)
 		return 2
 
+	if(mind?.martial_art?.deflection_chance) //Some martial arts users can deflect projectiles!
+		if(!lying && !(HULK in mutations) && prob(mind.martial_art.deflection_chance)) //But only if they're not lying down, and hulks can't do it
```

Comment:
Before, the prob check was on its own and did a `return ..()` immediately if it failed. The behavior is now changed a bit.

![image](https://user-images.githubusercontent.com/42044220/104513152-d1f68380-55b4-11eb-836f-7be259d80f2f.png)

There is now a chance to run some of this other code if the prob fails.

Reply:
Before that code was run before the check was done... it was a bit odd. the `..()` in that picture called the old martial arts check version of `bullet_act`. Now it only runs the code when it actually hits the target

Reply:
Yeah you're right. It's one of these weird duplicate proc situations, hate those things. Your self review about the call order makes a lot more sense now.